### PR TITLE
Singularity Support in Makeflow

### PR DESCRIPTION
**Pull request by Nekel-Seyew**
Added the ability to wrap tasks in singularity.

### CODE REVIEW

File: makeflow/src/makeflow.c [hunk 1]
```diff
@@ -1757,10 +1772,14 @@ int main(int argc, char *argv[])
 
 	/* XXX better to write created files to log, then delete those listed in log. */
 	if (container_mode == CONTAINER_MODE_DOCKER) {
-		char *cmd = string_format("rm %s", CONTAINER_SH);
-		system(cmd);
-		free(cmd);
-	}
+            char *cmd = string_format("rm %s", CONTAINER_SH);
```
Comment (dthain) [inline makeflow/src/makeflow.c]: Simplify to unlink(CONTAINER_SH)

Comment (dthain) [inline makeflow/src/makeflow.c]: For consistency: CONTAINER_SH -> CONTAINER_DOCKER_SH

File: makeflow/src/dag.h [hunk 2]
```diff
@@ -32,6 +32,7 @@ struct dag {
 	struct set *export_vars;           /* List of variables with prefix export. (these are setenv'ed eventually). */
 	struct set *special_vars;          /* List of special variables, such as category, cores, memory, etc. */
 	category_mode_t allocation_mode;   /* One of CATEGORY_ALLOCATION_MODE_{FIXED,MAX_THROUGHTPUT,MIN_WASTE} */
+        int wrapped_in_container;          /* Indicates whether or not the current makeflow is wrapped in a container*/
```
Comment (dthain) [inline makeflow/src/dag.h]: As far as I can tell, this doesn't get used.
And, it's an alias for CONTAINER_MODE_NONE.
Can it be removed?

**Comment by dthain** [comment]
Apart from a few nitpicks, this is looking good.  Please address those, squash to one commit, and solve conflicts.

**Comment by Nekel-Seyew** [comment]
Note: this code requires the custom version of Singularity I modified and installed on the disc servers. I will update this code when singularity 2.2 that includes my changes is release. The difference is that I use the --change-home </path/to/home> switch I added, and the v2.2 uses -H </path/to/home>. V2.2 is in testing, soon to be released as of this writing (sept 30)

**Comment by dthain** [comment]
@Nekel-Seyew is this ready to merge?

**Comment by dthain** [comment]
I nearly forgot, please update `doc/makeflow.html` and `doc/man/makeflow.m4` appropriately for the new features.

**Pull request merged**